src/main/java/org/spongepowered/common/data/builder/manipulator/SpongeDataBuilder.java

```diff
@@ -22,47 +22,48 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.common.data.builder.manipulator.mutable.entity;
+package org.spongepowered.common.data.builder.manipulator;
 
+import org.spongepowered.api.data.DataContainer;
 import org.spongepowered.api.data.DataHolder;
 import org.spongepowered.api.data.DataView;
-import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.DataManipulator;
 import org.spongepowered.api.data.manipulator.DataManipulatorBuilder;
-import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableHorseData;
-import org.spongepowered.api.data.manipulator.mutable.entity.HorseData;
+import org.spongepowered.api.data.manipulator.ImmutableDataManipulator;
 import org.spongepowered.api.service.persistence.InvalidDataException;
-import org.spongepowered.common.data.manipulator.mutable.entity.SpongeHorseData;
-import org.spongepowered.common.data.processor.common.HorseUtils;
-import org.spongepowered.common.data.processor.data.entity.HorseDataProcessor;
+import org.spongepowered.common.data.DataProcessor;
+import org.spongepowered.common.util.ReflectionUtil;
 
 import java.util.Optional;
 
-public class HorseDataBuilder implements DataManipulatorBuilder<HorseData, ImmutableHorseData> {
+public class SpongeDataBuilder<M extends DataManipulator<M, I>, I extends ImmutableDataManipulator<I, M>> implements DataManipulatorBuilder<M, I> {
 
-    private HorseDataProcessor processor;
+    private final Class<? extends M> manipulatorClass;
+    private final DataProcessor<M, I> processor;
```

Comment:
@gabizou I understand for `build(DataView)`, but what about `createFrom(DataHolder)`? I believe mods are supposed to add their own `DataProcessor`s if they want Sponge compatibility.